src/mesido/esdl/asset_to_component_base.py

```diff
@@ -68,46 +68,82 @@ def get_internal_energy(asset_name, carrier):
             carrier.pressure * 1.0e5,
             str(NetworkSettings.NETWORK_TYPE_HYDROGEN).upper(),
         )
+    elif NetworkSettings.NETWORK_TYPE_HEAT in carrier.name:
+        internal_energy = cP.CoolProp.PropsSI(
+            "U",
+            "T",
+            273.15 + temperature,
+            "P",
+            1.0 * 1.0e5,  # TODO: defualt 1 bar pressure should be set for the carrier
+            "WATER",
+        )
     else:
         logger.warning(
-            f"Neither gas or hydrogen was used in the carrier " f"name of pipe {asset_name}"
+            f"Neither gas/hydrogen/heat was used in the carrier " f"name of pipe {asset_name}."
         )
         # TODO: resolve heating value (default value below) vs internal energy values (above)
         internal_energy = 46.0e6  # natural gas at about 1 bar [J/kg] heating value
     return internal_energy  # [J/kg]
 
 
-def get_density(asset_name, carrier):
+def get_energy_content(asset_name, carrier) -> float:
+    # Return the heating value
+    energy_content_j_kg = 0.0  # [J/kg]
+    density_kg_m3 = get_density(
+        asset_name, carrier, temperature_degrees_celsius=20.0, pressure_pa=1.0e5
+    ) / 1000.0
+    if str(NetworkSettings.NETWORK_TYPE_GAS).upper() in str(carrier.name).upper():
+        # Groningen gas: 31,68 MJ/m3 LCV
+        energy_content_j_kg = 31.68 * 10.0**6 / density_kg_m3  # LCV / lower heating value
+    elif str(NetworkSettings.NETWORK_TYPE_HYDROGEN).upper() in str(carrier.name).upper():
+        # this value can be lower / higher heating value depending on the case
+        energy_content_j_kg = 120.0 * 10.0**6 / density_kg_m3
+    else:
+        raise logger.error(
+            f"Neither gas/hydrogen was used in the carrier " f"name of pipe {asset_name}."
+        )
+    return energy_content_j_kg
+
+
+def get_density(asset_name, carrier, temperature_degrees_celsius=20.0, pressure_pa=None):
     # TODO: gas carrier temperature still needs to be resolved.
-    # The default of 20°C is also used in the head_loss_class. Thus, when updating ensure it
-    # is also updated in the head_loss_class.
-    temperature = 20.0
+    # The default for temperature_degrees_celsius=20.0, this should be the same as the value (20°C)
+    # used in the head_loss_class for the calculation of the friction factor
+    # (linked to _kinematic_viscosity). Thus, when updating the default value of
+    # temperature_degrees_celsius ensure it is also updated in the head_loss_class.
+    if pressure_pa is None:
+        if str(NetworkSettings.NETWORK_TYPE_HEAT).upper() in str(carrier.name).upper():
+            pressure_pa = 1.0e5  # 1bar
```

Comment:
Default pressure of 1 bar does not make sense. Pressures will always be higher, I know it doesn't change a lot in the density value but maybe take a higher value. Typically 16bar is somewhat of a maximum pressure assumed.

Reply:
The pressure really does not make a big difference. But I agree a larger value would make more sense. I have updated it to 16 bar